tst/EnergyPlus/unit/ICEngineElectricGenerator.unit.cc

```diff
@@ -0,0 +1,138 @@
+        "  BG Exhaust Temperature Curve,  !- Exhaust Temperature Curve Name",
+        "  0.00952329,              !- Coefficient 1 of U-Factor Times Area Curve",
+        "  0.9,                     !- Coefficient 2 of U-Factor Times Area Curve",
+        "  0.00000063,              !- Maximum Exhaust Flow per Unit of Power Output {(kg/s)/W}",
+        "  150,                     !- Design Minimum Exhaust Temperature {C}",
+        "  45500,                   !- Fuel Higher Heating Value {kJ/kg}",
+        "  0.0,                     !- Design Heat Recovery Water Flow Rate {m3/s}",
+        "  ,                        !- Heat Recovery Inlet Node Name",
+        "  ,                        !- Heat Recovery Outlet Node Name",
+        "  Diesel;                  !- Fuel Type",
+
+        "Curve:Quadratic,",
+        "  BG Shaft Power Curve,    !- Name",
+        "  0.09755,                 !- Coefficient1 Constant",
+        "  0.6318,                  !- Coefficient2 x",
+        "  -0.4165,                 !- Coefficient3 x**2",
+        "  0,                       !- Minimum Value of x",
+        "  1;                       !- Maximum Value of x",
+
+        "Curve:Quadratic,",
+        "  BG Recovery Jacket Heat Curve,  !- Name",
+        "  0.25,                    !- Coefficient1 Constant",
+        "  0,                       !- Coefficient2 x",
+        "  0,                       !- Coefficient3 x**2",
+        "  0,                       !- Minimum Value of x",
+        "  1;                       !- Maximum Value of x",
+
+        "Curve:Quadratic,",
+        "  BG Recovery Lube Heat Curve,  !- Name",
+        "  0.15,                    !- Coefficient1 Constant",
+        "  0,                       !- Coefficient2 x",
+        "  0,                       !- Coefficient3 x**2",
+        "  0,                       !- Minimum Value of x",
+        "  1;                       !- Maximum Value of x",
+
+        "Curve:Quadratic,",
+        "  BG Total Exhaust Energy Curve,  !- Name",
+        "  0.1,                     !- Coefficient1 Constant",
+        "  0,                       !- Coefficient2 x",
+        "  0,                       !- Coefficient3 x**2",
+        "  0,                       !- Minimum Value of x",
+        "  1;                       !- Maximum Value of x",
+
+        "Curve:Quadratic,",
+        "  BG Exhaust Temperature Curve,  !- Name",
+        "  425,                     !- Coefficient1 Constant",
+        "  0,                       !- Coefficient2 x",
+        "  0,                       !- Coefficient3 x**2",
+        "  0,                       !- Minimum Value of x",
+        "  1;                       !- Maximum Value of x",
+    });
+
+    ASSERT_TRUE(process_idf(idf_objects));
+
+    GetICEngineGeneratorInput();
+
+    EXPECT_EQ(DataGlobalConstants::FuelType, "Diesel");
+    EXPECT_FALSE(DataGlobalConstants::FuelTypeErrorsFound);
```

Comment:
Since we are taking the global variables back out, you won't be able to check this.  And that's OK.  In the GetInput routine, the code that is performing input validation should check the value returned from the fuel validation function, and decide what to do.  If an error occurs, the GetInput routine should wrap up and throw a fatal error.  In that case, you can test successful fuel entries just by calling GetInput on good values.  And then you can test bad fuel entries by using EXPECT_THROW.  You can find examples if you search for that text in the unit test code.